bbb/meta-bbb/recipes-core/trebletrouble/files/src/tone.c

```diff
@@ -0,0 +1,255 @@
+#include <stdio.h>
+#include <malloc.h>
+#include <math.h>
+
+#include "libfft.h"
+#include "tone.h"
+
+#define FFT_SIZE 15
+
+/*
+ * The audio format will be PCM.
+ *  
+ * -------------Data to be determined ----------------
+ * numChannels -  1 for mono, 2 for stereo, etc...
+ * sampleRate - samples per second (44100 or 96000)
+ * bitsPerSample - 8 or 16 or 32. The higher, the better the resolution.
+ * subChunk2Size - size of the actual sound data.
+ * chunkSize - the size of the sound data + size of the header.
+ *  
+ */
+
+
+/* ------------------------------------------------------ [ Section: Endians ] */
+int isBigEndian() {
+  int test = 1;
+  char *p = (char*)&test;
+
+  return p[0] == 0;
+
+}
+
+void reverseEndianness(const long long int size, void* value) {
+
+  int i;
+  char result[32];
+  for (i = 0; i < size; i++) {
+    result[i] = ((char*)value)[size-i-1];
+  }
+  for (i = 0; i < size; i++) {
+    ((char*)value)[i] = result[i];
+  }
+}
+
+void toBigEndian(const long long int size, void* value) {
+  
+  char needsFix = !( (1 && isBigEndian()) || (0 && !isBigEndian()) );
+  if (needsFix) {
+    reverseEndianness(size,value);
+  }
+}
+
+void toLittleEndian(const long long int size, void* value) {
+  char needsFix = !( (0 && isBigEndian()) || (1 && !isBigEndian()) );
+  if (needsFix) {
+    reverseEndianness(size,value);
+  }
+}
+
+/* ------------------------------------------------------ [ Section: Wave Header ]  */
+
+
+WaveHeader makeWaveHeader(int const sampleRate, short int const numChannels, short int const bitsPerSample ) {
```

Comment:
Passing all this data on the stack is a bad idea. Better to pass in a pointer and fill it. (see below for example)
Also, `const int` instead of `int const` and `const short` instead of `short int const`